core/Plugin/Manager.php

```diff
@@ -882,6 +883,43 @@ private function reloadActivatedPlugin($pluginName, $pluginsToPostPendingEventsT
             return $pluginsToPostPendingEventsTo;
         }
 
+        if (!$this->isPluginBundledWithCore($pluginName)
+            && $this->isPluginActivated('Marketplace')
+            && $this->isPluginActivated($pluginName)) {
+
+            $info = $newPlugin->getInformation();
+            $hasInternet = Config::getInstance()->General['enable_internet_features'];
+
+            if (!empty($info['price']['base']) && $hasInternet) {
```

Comment:
We check this only for premium features, when the feature is activated, and only when the instance has the marketplace (we wouldn't be able to get the Marketplace class otherwise etc) and internet features enabled.